src/engine/Filter.cpp

```diff
@@ -232,18 +299,146 @@ vector<RT>* Filter::computeFilterFixedValue(
                            return std::regex_search(entity.value(), self_regex);
                          },
                          res);
-      break;
+    } break;
+    case SparqlFilter::PREFIX: {
+      // remove the leading '^' symbol
+      std::string rhs = _rhs.substr(1);
+      size_t lowerBound = getIndex().getVocab().getValueIdForGE(rhs);
+      std::string upperBoundStr = rhs;
+      // TODO(florian): This only works for ascii but could break unicode
+      upperBoundStr[upperBoundStr.size() - 1]++;
```

Comment:
Sounds good. I guess the string sorting does something like casting to `unsigned char` because otherwise the order too would be platform defined, which sounds like a bad idea. From my experience on Linux `char` is signed on x86_64 (unless passing `-funsigned-char` to GCC/clang). On ARM and ARM64 (e.g. on my RaspberryPi) on the other hand `char` is always `unsigned`. Interestingly enough this isn't just a `typedef`